src/status_im2/contexts/wallet/events.cljs

```diff
@@ -4,34 +4,103 @@
+    [utils.i18n :as i18n]
     [utils.re-frame :as rf]
     [utils.security.core :as security]
     [utils.transforms :as types]))
 
-(rf/defn get-wallet-token
-  {:events [:wallet/get-wallet-token]}
-  [{:keys [db]}]
-  (let [params (map :address (:profile/wallet-accounts db))]
-    {:json-rpc/call [{:method     "wallet_getWalletToken"
-                      :params     [params]
-                      :on-success #(rf/dispatch [:wallet/get-wallet-token-success %])
-                      :on-error   (fn [error]
-                                    (log/info "failed to get wallet token"
-                                              {:event  :wallet/get-wallet-token
-                                               :error  error
-                                               :params params}))}]}))
-
-(rf/defn get-wallet-token-success
-  {:events [:wallet/get-wallet-token-success]}
-  [{:keys [db]} data]
-  {:db (assoc db
-              :wallet/tokens          data
-              :wallet/tokens-loading? false)})
+(rf/reg-event-fx :wallet/show-account-created-toast
+ (fn [{:keys [db]} [address]]
+   (let [{:keys [name]} (get-in db [:wallet :accounts address])]
+     {:db (update db :wallet dissoc :navigate-to-account :new-account?)
+      :fx [[:dispatch
+            [:toasts/upsert
+             {:id         :new-wallet-account-created
+              :icon       :i/correct
+              :icon-color colors/success-50
+              :text       (i18n/label :t/account-created {:name name})}]]]})))
+
+(rf/reg-event-fx :wallet/navigate-to-account
+ (fn [{:keys [db]} [address]]
+   (let [new-account? (get-in db [:wallet :new-account?])]
+     (cond-> {:db (assoc-in db [:wallet :current-viewing-account-address] address)
+              :fx [[:dispatch [:navigate-to :wallet-accounts address]]]}
+
+       new-account?
+       (update :fx conj [:dispatch [:wallet/show-account-created-toast address]])))))
+
+(rf/reg-event-fx :wallet/close-account-page
+ (fn [{:keys [db]}]
+   {:db (update db :wallet dissoc :current-viewing-account-address)
+    :fx [[:dispatch [:navigate-back]]]}))
+
+(rf/reg-event-fx :wallet/get-accounts-success
+ (fn [{:keys [db]} [accounts]]
+   (let [wallet-db           (get db :wallet)
+         new-account?        (:new-account? wallet-db)
+         navigate-to-account (:navigate-to-account wallet-db)]
+     (cond-> {:db (reduce (fn [db {:keys [address] :as account}]
+                            (assoc-in db [:wallet :accounts address] account))
+                          db
+                          (data-store/rpc->accounts accounts))
+              :fx [[:dispatch [:wallet/get-wallet-token]]]}
+
+       new-account?
+       (update :fx
+               conj
+               [:dispatch [:hide-bottom-sheet]]
+               [:dispatch-later
+                [{:dispatch [:navigate-back]
+                  :ms       100}
+                 {:dispatch [:wallet/navigate-to-account navigate-to-account]
+                  :ms       300}]])))))
```

Comment:
Why do we navigate back and forth?

Reply:
I added this workaround, as when a user successfully creates a new account, he will land on the (individual) account screen.

```
Wallet Home -> Create Account -> Account Screen
```

On navigating back, the user will land back at the `Create Account` screen and this would probably end as a cycle.


We can use the `:navigate-back-to` event to directly take the user to the wallet home, but I'm not sure this would always be the case. We'll see the entry points for the individual account screen as we progress in development/design and update if it's only one entry point (wallet home).